frame/contracts/src/lib.rs

```diff
@@ -563,31 +549,74 @@ decl_module! {
 			let origin = ensure_signed(origin)?;
 			let dest = T::Lookup::lookup(dest)?;
 			let mut gas_meter = GasMeter::new(gas_limit);
-
 			let result = Self::execute_wasm(origin, &mut gas_meter, |ctx, gas_meter| {
 				ctx.call(dest, value, gas_meter, data)
 			});
-			gas_meter.into_dispatch_result(result)
+			gas_meter.into_dispatch_result(result, T::WeightInfo::call())
 		}
 
-		/// Instantiates a new contract from the `code_hash` generated by `put_code`,
-		/// optionally transferring some balance.
+		/// Instantiates a new contract from the supplied `code` optionally transferring
+		/// some balance.
+		///
+		/// This is the only function that can deploy new code to the chain.
 		///
-		/// The supplied `salt` is used for contract address deriviation. See `fn contract_address`.
+		/// # Parameters
+		///
+		/// * `endowment`: The balance to transfer from the `origin` to the newly created contract.
+		/// * `gas_limit`: The gas limit enforced when executing the constructor.
+		/// * `code`: The contract code to deploy in raw bytes.
+		/// * `data`: The input data to pass to the contract constructor.
+		/// * `salt`: Used for the address deriviation. See [`Self::contract_address`].
 		///
 		/// Instantiation is executed as follows:
 		///
+		/// - The supplied `code` is instrumented, deployed, and a `code_hash` is created for that code.
+		/// - If the `code_hash` already exists on the chain the underlying `code` will be shared.
 		/// - The destination address is computed based on the sender, code_hash and the salt.
 		/// - The smart-contract account is created at the computed address.
-		/// - The `ctor_code` is executed in the context of the newly-created account. Buffer returned
-		///   after the execution is saved as the `code` of the account. That code will be invoked
-		///   upon any call received by this account.
-		/// - The contract is initialized.
+		/// - The `endowment` is transfered to the new account.
+		/// - The `deploy` function is executed in the context of the newly-created account.
```

Comment:
I really wish `deploy` was called `instantiate`

Reply:
Yeah bad naming is kind of a thing in substrate. But I don't think it is worth changing. It will break ink! and no user will really have contact with this name.

Reply:
Yeah I know I know, just expressing the anguish I feel every time I see it :cry: